cmake/LinuxPackaging.cmake

```diff
@@ -100,7 +100,7 @@ if(DEB_DETECT_DEPENDENCIES AND DPKG_CMD AND DPKGQ_CMD)
 		${CPACK_DEBIAN_PACKAGE_DEPENDS})
 else()
 	# assume everything is turned on, and running on a modern OS
-	set(CPACK_DEBIAN_PACKAGE_DEPENDS "libaio-dev (>= 0.3.109), libavahi-client-dev (>= 0.6.31), libavahi-common-dev (>= 0.6.31), libc6-dev (>= 2.19), libusb-1.0-0-dev (>= 2:1.0.17), libxml2-dev (>= 2.9.1)")
+	set(CPACK_DEBIAN_PACKAGE_DEPENDS "libaio (>= 0.3.109), libavahi-client (>= 0.6.31), libavahi-common (>= 0.6.31), libc6 (>= 2.19), libusb-1.0-0 (>= 2:1.0.17), libxml2 (>= 2.9.1)")
```

Comment:
This is obviously a follow up change, but can this be adopted so that there is only a dependency on those packages for which features were enabled? E.g. if libiio is build without usb backend no libusb dependency.

Reply:
so, `CPACK_DEBIAN_PACKAGE_DEPENDS` is already being dynamically created on the `if()` branch ; this is the `else()` branch where things are added as fall-back;

for RPMs (and CentOS RPMs specifically), we surely could also do this; i just decided to go with a static list for now, and take a package list from CentOS 6, since those are currently the oldest versions that are being used for build;

Reply:
Hm, but we should always know which backends are enabled.

Reply:
That's true.
tbh, I don't know why this has been moved under a `DEB_DETECT_DEPENDENCIES` external variable (with default OFF).
We could remove it and always have that behavior.

No opinion

Reply:
Yes

Reply:
The DEB_DETECT_DEPENDENCIES is there to detect dependencies when others build and release binaries. We are not the only ones releasing debs.

Reply:
There may be more context-info that I am missing here about the `DEB_DETECT_DEPENDENCIES` part. So, I propose to defer this discussion on a PR that deals with this specifically.